product_multi_ean/models/product_ean.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
+from openerp import models, fields, api, exceptions, _
+from openerp.addons.product import product as addons_product
+
+
+class ProductEan13(models.Model):
+    _name = 'product.ean13'
+    _description = "List of EAN13 for a product."
+    _order = 'sequence'
+
+    name = fields.Char(string='EAN13', size=13, required=True)
+    sequence = fields.Integer(string='Sequence')
+    product_id = fields.Many2one(
+        string='Product',
+        comodel_name='product.product',
+        required=True)
+
+    @api.onchange('name')
+    def onchange_name(self):
+        if not addons_product.check_ean(self.name):
+            raise exceptions.Warning(_(
+                'You provided an invalid "EAN13 Barcode" reference. You may '
+                'use the "Internal Reference" field instead.'))
+
+    @api.one
+    @api.constrains('name')
+    def _check_name(self):
+        if not addons_product.check_ean(self.name):
+            raise exceptions.Warning(_(
+                'You provided an invalid "EAN13 Barcode" reference. You may '
+                'use the "Internal Reference" field instead.'))
+        eans = self.search([('id', '!=', self.id), ('name', '=', self.name)])
+        if eans:
+            raise exceptions.Warning(_(
+                'The EAN13 Barcode "%s" already exists for product "%s"!') % (
+                    self.name, eans[0].product_id.name))
+
+    def _auto_init(self, cr, context=None):
+        exist = self._table_exist(cr)
+        res = super(ProductEan13, self)._auto_init(cr, context=context)
+        if not exist:
+            cr.execute('INSERT INTO %s (product_id, name, sequence) '
+                       'SELECT id, ean13, 0 '
+                       'FROM product_product '
+                       'WHERE ean13 != \'\'' % self._table)
+        return res
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    ean13_ids = fields.One2many(
+        comodel_name='product.ean13',
+        inverse_name='product_id',
+        string='EAN13')
+    ean13 = fields.Char(
+        string='Main EAN13',
+        compute='_compute_ean13',
+        store=True)
+
+    @api.one
+    @api.depends('ean13_ids')
+    def _compute_ean13(self):
+        if self.ean13_ids:
+            if self.ean13 != self.ean13_ids[0]:
```

Comment:
This compares a string to a browse record and should probably read ``if self.ean13 != self.ean13_ids[0].name``.